File: L1/decompiled/3.c

undefined4 main(void)

{
  undefined4 local_8;
  
  printf("IOLI Crackme Level 0x03\n");
  printf("Password: ");
  scanf("%d",&local_8);
  test(local_8,0x52b24);
  return 0;
}

void test(int param_1,int param_2)

{
  if (param_1 == param_2) {
    shift("Sdvvzrug#RN$$$#=,");
  }
  else {
    shift("Lqydolg#Sdvvzrug$");
  }
  return;
}
void shift(char *param_1)

{
  size_t sVar1; 
  uint local_80; 
  char local_7c [120]; 
  
  local_80 = 0;
  while( true ) {
    sVar1 = strlen(param_1); //length of a string 
    if (sVar1 <= local_80) break; //if length of string is less than or equal to index, break
    local_7c[local_80] = param_1[local_80] + -3;  //shift each character by -3
    local_80 = local_80 + 1;
  }
  // so local_7c is the shifted string 
  local_7c[local_80] = '\0'; //null terminate the string
  printf("%s\n",local_7c); 
  return;
}

File: L1/decompiled/3.py
def shift(s: str):
    """Shift each character in the string by -3 and print the result."""
    shifted = ''.join(chr(ord(c) - 3) for c in s)
    print(shifted)


def test(input_val: int, correct_val: int):
    """Compare input to the correct value and print the shifted message."""
    if input_val == correct_val:
        shift("Sdvvzrug#RN$$$#=,")
    else:
        shift("Lqydolg#Sdvvzrug$")


def main():
    print("IOLI Crackme Level 0x03")
    password = int(input("Password: "))
    test(password, 0x52B24)  # 0x52B24 = 338724 in decimal


if __name__ == "__main__":
    main()
